src/invidious/playlists.cr

```diff
@@ -307,23 +307,32 @@ def subscribe_playlist(db, user, playlist)
   return playlist
 end
 
-def produce_playlist_url(id, index)
+def produce_playlist_continuation(id, index)
   if id.starts_with? "UC"
     id = "UU" + id.lchop("UC")
   end
   plid = "VL" + id
 
+  # Emulate a "request counter" increment, to make perfectly valid
+  # ctokens, even if at the time of writing, it's ignored by youtube.
+  request_count = (index / 100).to_i64 || 1_i64
+
   data = {"1:varint" => index.to_i64}
     .try { |i| Protodec::Any.cast_json(i) }
     .try { |i| Protodec::Any.from_json(i) }
     .try { |i| Base64.urlsafe_encode(i, padding: false) }
 
+  data_wrapper = {"1:varint" => request_count, "15:string" => "PT:#{data}"}
```

Comment:
Why did you change to this instead of using ```base64```?  That was made to do exactly this, though its not exactly documented.

https://github.com/omarroth/protodec/blob/master/src/protodec/utils.cr#L198

Reply:
I did that because "base64" doesn't allow the data to be URL-encoded (i.e with "=" replaced with "%3D").

I did not try to understand why it was done this way, I just tried to reproduce the YT token generated by youtube. That was the only way that worked.